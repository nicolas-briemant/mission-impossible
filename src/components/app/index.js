export { default as Title } from './title';
export { default as Menu } from './menu';
export { Header, HeaderLeft, HeaderRight } from './header';
