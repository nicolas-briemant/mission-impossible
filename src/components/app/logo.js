import React from 'react';
import glamorous from 'glamorous';

const LogoHeader = glamorous.img({
  verticalAlign: 'middle',
  width: '50px',
});

export default () => (
  <LogoHeader src="https://pbs.twimg.com/profile_images/576032932810227712/V-AdqQ_j_400x400.png" alt="logo" />
);
