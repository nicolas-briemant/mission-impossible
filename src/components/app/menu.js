import React from 'react';
import glamorous from 'glamorous';
import { Menu, MenuItem } from '@blueprintjs/core';
import 'normalize.css/normalize.css';
import '@blueprintjs/core/dist/blueprint.css';

const StyleMenu = glamorous(Menu)({
  display: 'flex',
});

export default () => (
  <StyleMenu>
    <MenuItem iconName="arrow-right" text="Missions" />
    <MenuItem iconName="people" text="People" />
  </StyleMenu>
);
