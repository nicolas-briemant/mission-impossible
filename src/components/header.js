import React from 'react';
import PropTypes from 'prop-types';
import { Navbar, NavbarGroup, Classes } from "@blueprintjs/core";
import cx from 'classnames';
import glamorous from 'glamorous';

export const HeaderLeft = ({ children }) => <NavbarGroup aligh="left">{children}</NavbarGroup>;
HeaderLeft.propTypes = { children: PropTypes.node };

export const HeaderRight = ({ children }) => <NavbarGroup align="right">{children}</NavbarGroup>;
HeaderRight.propTypes = { children: PropTypes.node };

const StyledHeader = glamorous(Navbar)(
  {},
  ({ isfixed }) => ({ zIndex: isfixed === 'true' ? 10 : 0 })
);

export const Header = ({ isFixed, children }) => {
  const headerLeft = () => React.Children.toArray(children).find((child) => child.type === HeaderLeft);
  const headerRight = () => React.Children.toArray(children).find(({type}) => type === HeaderRight);

  return (
    <StyledHeader
      isfixed={isFixed ? 'true' : 'false'}
      className={cx({[Classes.FIXED_TOP]: isFixed})}
    >
      {headerLeft()}
      {headerRight()}
    </StyledHeader>
  );
};

Header.propTypes = {
  isFixed: PropTypes.bool,
  children: PropTypes.node,
};
