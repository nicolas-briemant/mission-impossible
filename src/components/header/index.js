import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';

export const HeaderLeft = ({ children }) => <Fragment>{children}</Fragment>;

HeaderLeft.propTypes = {
  children: PropTypes.node,
};

export const HeaderRight = ({ children }) => <Fragment>{children}</Fragment>;

HeaderRight.propTypes = {
  children: PropTypes.node,
};

const HeaderContain = glamorous.div({
  position: 'relative',
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
});

export const Header = ({ children }) => {
  const headerLeft = () => React.Children.toArray(children).find(({ type }) => type === HeaderLeft);
  const headerRight = () => React.Children.toArray(children).find(({ type }) => type === HeaderRight);
  return (
    <HeaderContain>
      <div>{headerLeft()}</div>
      <div>{headerRight()}</div>
    </HeaderContain>
  );
};

Header.propTypes = {
  children: PropTypes.node,
};
