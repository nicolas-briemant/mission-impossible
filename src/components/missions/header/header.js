import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import { Navbar, NavbarGroup } from '@blueprintjs/core';
import 'normalize.css/normalize.css';

const MyNavbarGroup = glamorous(NavbarGroup)({
  display: 'flex',
  alignItems: 'center',
});

export const HeaderLeft = ({ children }) => <Fragment>{children}</Fragment>;

HeaderLeft.propTypes = {
  children: PropTypes.node,
};

export const HeaderRight = ({ children }) => <Fragment>{children}</Fragment>;

HeaderRight.propTypes = {
  children: PropTypes.node,
};

export const Header = ({ children }) => {
  const headerLeft = () => React.Children.toArray(children).find(({ type }) => type === HeaderLeft);
  const headerRight = () => React.Children.toArray(children).find(({ type }) => type === HeaderRight);

  return (
    <header>
      <Navbar>
        <MyNavbarGroup align="left">{headerLeft()}</MyNavbarGroup>
        <MyNavbarGroup align="right">{headerRight()}</MyNavbarGroup>
      </Navbar>
    </header>
  );
};

Header.propTypes = {
  children: PropTypes.node,
};
