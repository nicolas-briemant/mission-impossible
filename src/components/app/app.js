import React from 'react';
import { connect } from 'react-redux';
import './app.css';
import Missions from '../missions';
import { Header, HeaderLeft, HeaderRight } from '../header';
import { removeMission, selectMission, removeSelectedMissions, blockedAction } from '../../actions';
import Logo from '../header/logo';
import HeaderMenu from '../header/menu';
import logo from '../../img/007.png';

const App = props => (
  <div className="App">
    <Header>
      <HeaderLeft>
        <Logo src={logo} alt="007" title="Missions Impossible" />
      </HeaderLeft>
      <HeaderRight>
        <HeaderMenu {...props} />
      </HeaderRight>
    </Header>
    <Missions {...props} />
  </div>
);

const mapDisPatchToProps = { removeMission, selectMission, removeSelectedMissions, blockedAction };

const mapStateToProps = state => state;

export default connect(mapStateToProps, mapDisPatchToProps)(App);
