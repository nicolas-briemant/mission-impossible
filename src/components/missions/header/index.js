import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import { ButtonGroup, Button } from '@blueprintjs/core';
import { Header, HeaderLeft, HeaderRight } from './header';
import logo from '../../../data/logo.png';

const Logo = glamorous.img({
  width: 75,
});

const Header2 = ({ blockAction }) => (
  <Header>
    <HeaderLeft>
      <Logo alt="Logo" src={logo} />
      <h4>Titre du site</h4>
    </HeaderLeft>
    <HeaderRight>
      <ButtonGroup minimal="true" large="true">
        <Button iconName="pt-icon-align-justify" onClick={() => blockAction()}>
          Missions
        </Button>
        <Button iconName="people" onClick={() => blockAction()}>
          Peoples
        </Button>
      </ButtonGroup>
    </HeaderRight>
  </Header>
);

Header2.propTypes = {
  blockAction: PropTypes.func,
};

export default Header2;
