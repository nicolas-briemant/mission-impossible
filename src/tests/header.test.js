import React from 'react';
import { shallow, mount } from 'enzyme';
import { Header, HeaderLeft, HeaderRight } from '../components/app/header';

describe('Header', () => {
  it('should match snapchot', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapchot with children', () => {
    const root = (
      <Header>
        <HeaderLeft>
          <span>left</span>
        </HeaderLeft>
      </Header>
    );
    const wrapper = mount(root);
    expect(wrapper.find(HeaderLeft)).toHaveLength(1);
  });

  it('should match snapchot with children text', () => {
    const root = (
      <Header>
        <HeaderLeft>
          <span>left</span>
        </HeaderLeft>
        <HeaderRight>
          <span>right</span>
        </HeaderRight>
      </Header>
    );
    const wrapper = mount(root);
    const expectedElements = [<span>left</span>, <span>right</span>];
    expect(wrapper.containsAllMatchingElements(expectedElements)).toEqual(true);
  });
});
