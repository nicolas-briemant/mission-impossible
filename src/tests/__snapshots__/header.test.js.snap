// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapchot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
      ],
      "className": "",
      "isfixed": "false",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
        ],
        "className": "",
        "isfixed": "false",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
